java/src/jmri/implementation/MatrixSignalMast.java

```diff
@@ -0,0 +1,444 @@
+        outputlist.add(outputsToBeans.get("output1").getName()); // convert NBH to name (String)
+        if (outputsToBeans.containsKey("output2")) {
+            // outputsToBeans hashmap
+            outputlist.add(outputsToBeans.get("output2").getName());
+        }
+        if (outputsToBeans.containsKey("output3")) {
+            outputlist.add(outputsToBeans.get("output3").getName());
+        }
+        if (outputsToBeans.containsKey("output4")) {
+            outputlist.add(outputsToBeans.get("output4").getName());
+        }
+        if (outputsToBeans.containsKey("output5")) {
+            outputlist.add(outputsToBeans.get("output5").getName());
+        }
+        return outputlist;
+    }
+
+    HashMap<String, NamedBeanHandle<Turnout>> outputsToBeans = new HashMap<String, NamedBeanHandle<Turnout>>(5); // output# - bean pairs
+
+    //looks a lot like the next method, remove?
+    NamedBeanHandle<Turnout> TurnoutNameToHandle (String turnoutName) {
+        if (turnoutName != null && !turnoutName.equals("")) {
+            Turnout turn = jmri.InstanceManager.turnoutManagerInstance().getTurnout(turnoutName);
+            NamedBeanHandle<Turnout> namedTurnout = jmri.InstanceManager.getDefault(jmri.NamedBeanHandleManager.class).getNamedBeanHandle(turnoutName, turn);
+            return namedTurnout;
+        }
+        log.error("Trying to create an empty output");
+        return null;
+    }
+
+    /**
+     * Receive properties from xml, convert name to NamedBeanHandle, store in hashmap outputsToBeans
+     * @param colname String describing the name of the corresponding output, like "output1"
+     * @param turnoutname String for the display name of the output, like "LT1"
+    */
+    public void setOutput(String colname, String turnoutname) {
+        if (colname == null || turnoutname == null){
+            log.error("Trying to set a null output. Fix output configuration for mast");
+        } else {
+            Turnout turn = jmri.InstanceManager.turnoutManagerInstance().getTurnout(turnoutname);
+            NamedBeanHandle<Turnout> namedTurnout = jmri.InstanceManager.getDefault(jmri.NamedBeanHandleManager.class).getNamedBeanHandle(turnoutname, turn);
+            if (outputsToBeans.containsKey(colname)) {
+                log.debug("Output " + colname + " is already defined so will override");
+                outputsToBeans.remove(colname);
+            }
+            outputsToBeans.put(colname, namedTurnout);
+        }
+    }
+
+    /**
+     *  Send hardware instruction
+     *  @param bits char[] of on/off outputs per aspect, like "00010"
+     *  length of array should match the number of outputs defined
+     */
+    public void updateOutputs (char[] bits) {
+        if (bits == null){
+            log.debug("Empty char[] received");
+        }
+        for (int i = 0; i < BitNum; i++) {
+            //if (bits[i] == '1' && getTurnoutBean(i).getCommandedState() == Turnout.THROWN) { // no need to set a state already set
```

Comment:
In the Signal Mast Panel, when I select an Aspect for a mast with this driver, the drop down doesn't pop back, and the Edit button no longer responds. Turnouts are present. The Console says:
``2016-05-24 07:40:27,743 ptionhandler.UncaughtExceptionHandler ERROR - Uncaught Exception caught by jmri.util.exceptionhandler.UncaughtExceptionHandler [AWT-EventQueue-0]
java.lang.NullPointerException
	at jmri.implementation.MatrixSignalMast.getTurnoutBean(MatrixSignalMast.java:363)
	at jmri.implementation.MatrixSignalMast.updateOutputs(MatrixSignalMast.java:334)
	at jmri.implementation.MatrixSignalMast.setAspect(MatrixSignalMast.java:139)
	at jmri.jmrit.beantable.signalmast.SignalMastTableDataModel.setValueAt(SignalMastTableDataModel.java:173)
	at jmri.util.com.sun.TableSorter.setValueAt(TableSorter.java:320)
	at javax.swing.JTable.setValueAt(JTable.java:2741)
	at javax.swing.JTable.editingStopped(JTable.java:4723)
	at javax.swing.AbstractCellEditor.fireEditingStopped(AbstractCellEditor.java:141)
	at javax.swing.DefaultCellEditor$EditorDelegate.stopCellEditing(DefaultCellEditor.java:368)
	at javax.swing.DefaultCellEditor$3.stopCellEditing(DefaultCellEditor.java:158)
	at javax.swing.DefaultCellEditor.stopCellEditing(DefaultCellEditor.java:233)
	at javax.swing.DefaultCellEditor$EditorDelegate.actionPerformed(DefaultCellEditor.java:385)
	at javax.swing.JComboBox.fireActionEvent(JComboBox.java:1258)
	at javax.swing.JComboBox.setSelectedItem(JComboBox.java:586)
	at javax.swing.JComboBox.setSelectedIndex(JComboBox.java:622)
	at javax.swing.plaf.basic.BasicComboPopup$Handler.mouseReleased(BasicComboPopup.java:852)
	at java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:290)
	at java.awt.Component.processMouseEvent(Component.java:6535)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at com.apple.laf.AquaComboBoxPopup$1.processMouseEvent(AquaComboBoxPopup.java:157)
	at java.awt.Component.processEvent(Component.java:6300)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4891)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4713)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4525)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
	at java.awt.Container.dispatchEventImpl(Container.java:2280)
	at java.awt.Window.dispatchEventImpl(Window.java:2750)
	at java.awt.Component.dispatchEvent(Component.java:4713)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)``